### PR TITLE
Fix MAVStation USB bootloader

### PR DESCRIPTION
**Pull request by ksschwabe**
The MAVStation USB bootloader was accidentally killed by the introduction
of the dual USB and USART bootloader.

### CODE REVIEW

**Comment by ksschwabe** [comment]
@davids5 : It looks as if I accidentally killed the USB bootloader for the MAVStation board. I introduced a INTEFACE_USART into the hw_config.h file for it and that meant that it always would initialise the USART interface and not the USB interface. 

Since the MAV Station can have a bootloader up to 0x3000 bytes (~12kB), I have set it to both the USB and USART bootloader capability. The only thing that I am not sure about is whether this means I need to do something differently with how its clock is set up (see my line comment).

File: main_f1.c [hunk 1]
```diff
@@ -15,16 +15,12 @@
...
-# define BOARD_INTERFACE_CONFIG		(void *)BOARD_USART
-#else
-# define BOARD_INTERFACE_CONFIG		NULL
+/* context passed to cinit */
+#if INTERFACE_USART
+# define BOARD_INTERFACE_CONFIG_USART	(void *)BOARD_USART
+#endif
+#if INTERFACE_USB
+# define BOARD_INTERFACE_CONFIG_USB  	NULL
 #endif
```
Comment (ksschwabe) [inline main_f1.c]: Owing to my addition of the INTERFACE_USART in the hw_config.h file, this resulted in only teh USART interface being set.

File: main_f1.c [hunk 2]
```diff
@@ -128,7 +124,7 @@ board_deinit(void)
 static inline void
 clock_init(void)
 {
-#if defined(INTERFACE_USB)
+#if INTERFACE_USB
 	rcc_clock_setup_in_hsi_out_48mhz();
 #else
 	rcc_clock_setup_in_hsi_out_24mhz();
```
Comment (ksschwabe) [inline main_f1.c]: I am not sure if I need to change the clock setup for the MAV Station. As far as I know, it should also be able to use the USART BL if the clock is setup for the USB.

Comment (davids5) [inline main_f1.c]: I agree it makes sense to always use ```rcc_clock_setup_in_hsi_out_48mhz();```. But a good exercise would be to look at the commit that added the mavstation and see if there are any other implications on choice of frequency.

**Comment by ksschwabe** [comment]
@kbrtny : I accidentally killed the MAVStation bootloader. I have also noticed the USART port is not correctly configured and no one noticed because it was only using the USB for bootloading.

I saw that you did the refactoring of the bootloader for the MAVStation, hence my asking. Is there any reason to use a different clock frequency for the USB as opposed to the USART bootloaders? Or can I just use the faster frequency for both? Also do you even want the ability to bootload from the USART port in addition to the USB port?

**Comment by davids5** [comment]
Closing this as Mavstation has EOL and is not supported.

**Pull request closed without merging**